### PR TITLE
feat: provide validation properties and throw more descriptive exceptions

### LINKED ISSUE #22: Documentation/API improvement: SceneReference is serialized so we must check AssetGuidHex not empty, else need IsValid method
I wanted to check if SceneReference was null before using it, but due to serialization, Unity creates a default instance with invalid fields (sceneAsset is null, strings are empty, other properties cause exception on access).

So, basically, my null check `if (mySceneReference != null)` is always verified, and my code tries to use it even when not set.

Trying to access BuildIndex then causes an obscure error:

> KeyNotFoundException: The given key '' was not present in the dictionary.
System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) (at <7560840559c84ffbb83a08c4696f01f7>:0)
Eflatun.SceneReference.SceneReference.get_Path () (at Library/PackageCache/com.eflatun.scenereference@1.2.2/Runtime/SceneReference.cs:26)
Eflatun.SceneReference.SceneReference.get_BuildIndex () (at Library/PackageCache/com.eflatun.scenereference@1.2.2/Runtime/SceneReference.cs:31)

There's not much to do about it if we want to serialize the whole class, but at least the README documentation should indicate in usage that before using the field, it may be good to check for `if (!string.IsNullOrEmpty(mySceneReference.AssetGuidHex)` or the likes.

I also tried to check `mySceneReference.sceneAsset != null`, but the field is internal so I could not access it. I suppose we could add a `bool IsValid()` method that checks sceneAsset to avoid the more complicated AssetGuidHex check.

### PR DESCRIPTION
**Pull request by starikcetin**
Closes #22

### CODE REVIEW

File: Eflatun.SceneReference/Packages/com.eflatun.scenereference/Runtime/SceneReference.cs [hunk 1]
```diff
@@ -12,32 +12,143 @@ namespace Eflatun.SceneReference
...
+        /// If you only need to check if it is safe to use this <see cref="SceneReference"/>, then only check <see cref="IsSafeToUse"/> instead. Checking only <see cref="IsSafeToUse"/> is sufficient for the majority of the use cases.
+        /// </remarks>
+        /// <seealso cref="IsInSceneGuidToPathMap"/>
+        /// <seealso cref="IsInBuildAndEnabled"/>
+        /// <seealso cref="IsSafeToUse"/>
+        public bool HasValue
+        {
+            get
+            {
+                if (string.IsNullOrWhiteSpace(AssetGuidHex))
```
Comment (starikcetin) [inline Eflatun.SceneReference/Packages/com.eflatun.scenereference/Runtime/SceneReference.cs:91]: Also check if 32 chars. Don't forget to update the internal exception.

Comment (starikcetin) [reply Eflatun.SceneReference/Packages/com.eflatun.scenereference/Runtime/SceneReference.cs]: Done in 16f5925f462869e2bb337760b48f380662d2209c

**Comment by starikcetin** [comment]
Included in [`1.3.0`](https://github.com/starikcetin/Eflatun.SceneReference/releases/tag/1.3.0).

**Comment by hsandt** [comment]
Thanks, trying it now. To confirm, IsValidSceneAsset (which was added in intermediate commits) has been removed for the final PR, which now suggests to use IsSafeToUse in most cases, as it combines HasValue/IsInSceneGuidToPathMap/IsInBuildAndEnabled in one, correct?

**Comment by starikcetin** [comment]
> Thanks, trying it now. To confirm, IsValidSceneAsset (which was added in intermediate commits) has been removed for the final PR, which now suggests to use IsSafeToUse in most cases, as it combines HasValue/IsInSceneGuidToPathMap/IsInBuildAndEnabled in one, correct?

Yes, `IsSafeToUse` is enough on its own to tell you that the `SceneReference` is completely valid ([see docs here](https://github.com/starikcetin/Eflatun.SceneReference#validation)). All other validation properties are for rare edge cases where you need partial validation ([see docs here](https://github.com/starikcetin/Eflatun.SceneReference#partial-validation)).

**Pull request merged**